test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,390 @@
+
+    writeln(foo); // Outputs 12
+    writeln(bar); // Outputs 5 after output in Conflict.bar function
+    writeln(other); // Outputs 5.0 + 3.0i
+  }
+
+
+  {
+    /* If both symbols which conflict are desired, or if the use causes symbols
+       to be shadowed which are necessary, you can choose to rename a symbol
+       when including it, so long as the new name does not cause any conflicts
+       with other included symbols.
+    */
+    use modToUse;
+    use Conflict only bar as boop;
+    writeln(bar); // Outputs 2
+    writeln(boop); // Outputs 5 after output in Conflict.bar function
+  }
+
+  writeln();
+  writeln("Application to enums");
+
+  {
+    /* 'Use' statements can also be called on enums.  Normally to access one of
+       an enum's constants, you must provide a prefix of the enum name.  With a
+       'use' of that enum, such a prefix is no longer necessary.
+    */
+
+    enum color {red, blue, yellow};
+
+    // Normally you must use the enum name as a prefix
+    var aColor = color.blue;
+    writeln(aColor);
+
+    use color;
+
+    // The use statement allows you to access it without the prefix
+    var anotherColor = yellow; // color.yellow
+    writeln(anotherColor);
+
+    // All of the above rules for 'use' statements also apply to 'use's of enums
+  }
+
+  writeln();
+  writeln("Accessing modules from other files");
+
+  {
+    /* Modules that live outside of the file sometimes have special rules for
+       how to access them.  If the module you wish to access has the same name
+       as the file in which it lives, and this file is in the same directory as
+       your program, no additional steps are necessary.
+    */
+
+    use modulesHelper;
+
+    writeln(someVar); // Should be 19
+  }
+
+
+  {
+    /* When the module you wish to access is not named after the file in which
+       it lives, you must name the file as part of the compilation step:
+
+       chpl modules.chpl modulesHelper2.chpl
+    */
+
+    use anotherHelper;
+
+    writeln(someFunc()); // Should be 23
+
+    /* Note, the above will only compile if all the code after this closing
+       brace until the end of the main() function is removed or commented out.
+       A summary compilation line will be provided at the end
+    */
+  }
+
+
+  {
+    /* If the helper module also defines a main() function, or this module does
+       not, then you must specify on the command line at compilation time which
+       module should be the main module for the program, using the --main-module
+       flag:
+
+       chpl modules.chpl modulesHelper3.chpl --main-module modules
+    */
+
+    use helperWithMain;
+
+    writeln(someVar); // Should be 19
+
+    /* Note, the above will only compile if all the code after this closing
+       brace until the end of the main() function is removed or commented out,
+       as well as the code for accessing anotherHelper.  A summary compilation
+       line will be provided at the end
+    */
+  }
```

Comment:
I might be tempted to remove everything from the enum example to here, as it feels less about how the language is defined and more about the quirks of our compiler.  E.g., we've talked about really wanting a compiler that will scan all source files in the module search path to find which modules they define without requiring the filename to match or be listed on the command-line, and I think that's the model we'd like for Chapel.  The above is practical info, but feels almost orthogonal to the namespace control concerns that modules strive to provide and that I think of this primer as trying go teach.